celesta-sql/src/main/java/ru/curs/celesta/score/Grain.java

```diff
@@ -547,8 +459,36 @@ void addNativeSql(String sql, boolean isBefore, DBType dbType, GrainPart grainPa
         return afterSql.getOrDefault(dbType, Collections.emptyList());
     }
 
-    public Set<GrainPart> getGrainParts() {
+    public Set<? extends GrainPart> getGrainParts() {
```

Comment:
What was the problem here? `grainParts` has ` Set<GrainPart>` type, returning wildcard types should be avoided. Do we really need this?

Reply:
Yes, the goal here was to get rid of the **set** modification over getter like:
`grain.getGrainParts().add(...)`, since changing collections in this way may be fragile (the incapsulated collection could become f.e. unmodifiable in the future). Therefore `addGrainPart()` was excplicitly added to the Grain API. I'll remove the wilcard though.

Reply:
Wow, I understood the idea! Never thought about making collections immutable this way. 🤔  This, however, will not prevent anyone from calling `Set.remove(Object o)`. 

So... either `Collections.immutableSet` or just let's give it up and let's hope misusing this getter is misusers' problem.

Reply:
I did change back to `public Set<GrainPart> getGrainParts()`